src/components/modules/document/DocRequiredCreation/modules/UploadRequired/UploadRequired.interactor.vue

```diff
@@ -27,11 +26,18 @@ export default {
       this.$emit('cancel');
     },
 
-    onConfirm({ documentId, status }) {
-      this.$options.documentsRequiredStore.addDocTypeStatus({
-        documentType: this.selectedDocumentType,
-        id: documentId,
+    async onNext({ documentId, status, dateOfExpiry }) {
+      const { selectedDocumentType: documentType } = this;
+      await this.gateway.addDocTypeStatus({
+        documentType,
+        documentId,
         status,
+        dateOfExpiry,
+      });
+
+      await this.gateway.selectDocumentForType({
```

Comment:
Can we use `Promise.all`?

Reply:
nope, they should be called step by step